### PR TITLE
migrate wallet, walletmanager to events

### PR DESCRIPTION
**Pull request by wolverineks**
fixes [YOMO-327](https://emurgo.atlassian.net/browse/YOMO-327)

### CODE REVIEW

File: src/Settings/SettingsScreenNavigator.tsx [hunk 1]
```diff
@@ -93,8 +93,14 @@ export const SettingsScreenNavigator = () => {
       />
 
       <Stack.Screen //
-        name="easy-confirmation"
-        component={ToggleEasyConfirmationScreen}
+        name="enable-easy-confirmation"
+        component={EnableEasyConfirmationScreen}
+        options={{title: strings.toggleEachConfirmationTitle}}
```
Comment (stackchain) [inline src/Settings/SettingsScreenNavigator.tsx]: typo

```suggestion
        options={{title: strings.toggleEasyConfirmationTitle}}
```

File: src/Settings/SettingsScreenNavigator.tsx [hunk 2]
```diff
@@ -93,8 +93,14 @@ export const SettingsScreenNavigator = () => {
...
+      />
+
+      <Stack.Screen //
+        name="disable-easy-confirmation"
+        component={DisableEasyConfirmationScreen}
         options={{title: strings.toggleEachConfirmationTitle}}
```
Comment (stackchain) [inline src/Settings/SettingsScreenNavigator.tsx]: ⬆️

File: src/hooks/index.ts [hunk 3]
```diff
@@ -41,6 +42,59 @@ import {YoroiAmounts, YoroiSignedTx, YoroiUnsignedTx} from '../yoroi-wallets/typ
...
+
+    return () => {
+      unsubWallet()
+      unsubWalletManager()
+    }
+  }, [event, wallet])
+}
+
+export const useEnableEasyConfirmation = (
+  _wallet: YoroiWallet,
```
Comment (stackchain) [inline src/hooks/index.ts]: how about adding it after migrating from wallet manager?
```suggestion
```

File: src/hooks/index.ts [hunk 4]
```diff
@@ -41,6 +42,59 @@ import {YoroiAmounts, YoroiSignedTx, YoroiUnsignedTx} from '../yoroi-wallets/typ
...
+    mutationFn: ({password, intl}) => walletManager.enableEasyConfirmation(password, intl),
+  })
+
+  return {
+    ...mutation,
+    enableEasyConfirmation: mutation.mutate,
+  }
+}
+
+export const useDisableEasyConfirmation = (_wallet: YoroiWallet, options?: UseMutationOptions) => {
```
Comment (stackchain) [inline src/hooks/index.ts]: ⬆️

File: src/yoroi-wallets/Wallet.ts [hunk 5]
```diff
@@ -179,12 +191,16 @@ export class Wallet {
...
+    this.subscriptions.forEach((handler) => handler(event))
   }
 
-  subscribe(handler: (Wallet) => void) {
-    this._subscriptions.push(handler)
+  subscribe(subscription: WalletSubscription) {
+    this.subscriptions.push(subscription)
+
+    return () => {
+      this.subscriptions = this.subscriptions.filter((sub) => sub !== subscription)
```
Comment (stackchain) [inline src/yoroi-wallets/Wallet.ts:202]: 🚀

File: src/hooks/index.ts [hunk 6]
```diff
@@ -41,6 +42,59 @@ import {YoroiAmounts, YoroiSignedTx, YoroiUnsignedTx} from '../yoroi-wallets/typ
...
+  const mutation = useMutation({
+    ...options,
+    mutationFn: async () => walletManager.disableEasyConfirmation(),
```
Comment (stackchain) [inline src/hooks/index.ts:82]:
```suggestion
    mutationFn: () => walletManager.disableEasyConfirmation(),
```

**Review by stackchain — CHANGES_REQUESTED** [review]

File: src/Settings/EasyConfirmation/EasyConfirmationScreen.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import {storiesOf} from '@storybook/react-native'
+import React from 'react'
+
+import {mockWallet} from '../../../storybook'
+import {SelectedWalletProvider} from '../../SelectedWallet'
+import {DisableEasyConfirmationScreen} from './DisableEasyConfirmationScreen'
+import {EnableEasyConfirmationScreen} from './EnableEasyConfirmationScreen'
+
+storiesOf('EasyConfirmatio Screen', module)
```
Comment (banklesss) [inline src/Settings/EasyConfirmation/EasyConfirmationScreen.stories.tsx]: typo

Comment (stackchain) [reply src/Settings/EasyConfirmation/EasyConfirmationScreen.stories.tsx]: > typo

```suggestion
storiesOf('EasyConfirmation Screen', module)
```

File: src/Settings/EasyConfirmation/EnableEasyConfirmationScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+    },
+  })
+
+  React.useEffect(() => {
+    const unsubscribe = navigation.addListener('blur', () => {
+      clearPassword()
+    })
+    return unsubscribe
+  }, [navigation])
+
```
Comment (stackchain) [inline src/Settings/EasyConfirmation/EnableEasyConfirmationScreen.tsx]:
```suggestion
```

File: src/Settings/EasyConfirmation/EnableEasyConfirmationScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+import {isEmptyString} from '../../legacy/utils'
+import {useSelectedWalletMeta, useSetSelectedWalletMeta} from '../../SelectedWallet'
+import {COLORS} from '../../theme'
+
+export const EnableEasyConfirmationScreen = () => {
+  const intl = useIntl()
+  const strings = useStrings()
+  const navigation = useNavigation()
+  const [masterPassword, setMasterPassword] = React.useState('')
+  const clearPassword = () => setMasterPassword('')
```
Comment (stackchain) [inline src/Settings/EasyConfirmation/EnableEasyConfirmationScreen.tsx]:
```suggestion
```

**Review by stackchain — CHANGES_REQUESTED** [review]

**Review by stackchain — APPROVED** [review]

**Pull request merged**